### PR TITLE
Implement feature 5973 - add SARIF output format

### PR DESCRIPTION
**Pull request by llaville**
As it is my first PR on this project, hope I don't forget any steps.

- `make cs` OK
- `make tests` OK

My first remark:

- As I don't found any API / constant that give current PHPStan version, I've hardcoded it in `SarifErrorFormatter` and unit tests.

Waiting for your feedbacks

### CODE REVIEW

**Comment by llaville** [comment]
`make phpstan` has an error (even on master branch)

```
php bin/phpstan clear-result-cache -q && php -d memory_limit=768M bin/phpstan
Note: Using configuration file /shared/backups/github/my-phpstan-src/phpstan.neon.dist.
 1263/1263 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------------------------------------------
  Line   src/Rules/RuleErrorBuilder.php
 ------ ----------------------------------------------------------------------------------------------------------
         Ignored error pattern #^Variable property access on PHPStan\\Rules\\RuleError\.$# in path
         /shared/backups/github/my-phpstan-src/src/Rules/RuleErrorBuilder.php was not matched in reported errors.
  149    Variable property access on object.
  152    Method PHPStan\Rules\RuleErrorBuilder::build() should return PHPStan\Rules\RuleError but returns object.
 ------ ----------------------------------------------------------------------------------------------------------


 [ERROR] Found 3 errors


make: *** [Makefile:59: phpstan] Error 1
```

File: conf/bleedingEdge.neon [hunk 1]
```diff
@@ -2,3 +2,7 @@ parameters:
 	featureToggles:
 		bleedingEdge: true
 		skipCheckGenericClasses: []
+
+services:
+	errorFormatter.sarif:
+		class: PHPStan\Command\ErrorFormatter\SarifErrorFormatter
```
Comment (ondrejmirtes) [inline conf/bleedingEdge.neon:8]: This should be in `config.neon.`

File: composer.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
 	],
 	"require": {
 		"php": "^7.4 || ^8.0",
+		"bartlett/sarif-php-sdk": "^1.0",
```
Comment (ondrejmirtes) [inline composer.json:9]: Please try to do this without a dependency. It's just encoding a simple JSON after all.

File: src/Command/ErrorFormatter/SarifErrorFormatter.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ * @author Laurent Laville
+ */
+class SarifErrorFormatter implements ErrorFormatter
+{
+
+	public function formatErrors(AnalysisResult $analysisResult, Output $output): int
+	{
+		$driver = new ToolComponent('PHPStan');
+		$driver->setInformationUri('https://phpstan.org');
+		$driver->setVersion('1.1.2');
```
Comment (ondrejmirtes) [inline src/Command/ErrorFormatter/SarifErrorFormatter.php:30]: The version can be obtained like this: https://github.com/phpstan/phpstan-src/blob/1789915a3ed766e8e784ac3eabee4b327e9d192a/bin/phpstan#L109-L114

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
And please fix the Windows failures :)

**Comment by ondrejmirtes** [comment]
Feel free to open a new PR if you decide to finish this, thanks :)

**Pull request closed without merging**